### PR TITLE
Adding support to NettyPerfCient to target account and container

### PR DESCRIPTION
**Pull request by nsivabalan**
1. Exposing versions in Container
2. Adding support to NettyPerfCient to target account and container for POST.

### CODE REVIEW

File: ambry-api/src/test/java/com.github.ambry/account/InMemAccountService.java [hunk 1]
```diff
@@ -114,9 +114,10 @@ public synchronized void clear() {
    * Creates and adds an {@link Account} to this {@link AccountService}. The account will contain one container
    * with {@link Container#DEFAULT_PUBLIC_CONTAINER_ID}, one with {@link Container#DEFAULT_PRIVATE_CONTAINER_ID} and
    * one other random {@link Container}.
+   * @param enableEncryption {@code true} if encryption needs to be enabled. {@code false} otherwise
    * @return the {@link Account} that was created and added.
    */
-  public synchronized Account createAndAddRandomAccount() {
+  public synchronized Account createAndAddRandomAccount(boolean enableEncryption) {
```
Comment (vgkholla) [inline ambry-api/src/test/java/com.github.ambry/account/InMemAccountService.java]: don't do this - this is supposed to be a method that takes no params and makes it easy to just create a random account. Add whatever containers you want after.

File: ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java [hunk 2]
```diff
@@ -153,6 +159,23 @@ protected ClientArgs(String args[]) {
               .withOptionalArg()
               .describedAs("postBlobChunkSize")
               .ofType(Integer.class);
+      ArgumentAcceptingOptionSpec<String> targetAccountName =
+          parser.accepts("targetAccountName", "Target account name for POSTs")
+              .withOptionalArg()
+              .describedAs("targetAccountName")
+              .ofType(String.class)
+              .defaultsTo("ambry-frontend-test");
```
Comment (vgkholla) [inline ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java]: remove

File: ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java [hunk 3]
```diff
@@ -153,6 +159,23 @@ protected ClientArgs(String args[]) {
...
+          parser.accepts("targetContainerName", "Target container name for POSTs")
+              .withOptionalArg()
+              .describedAs("targetContainerName")
+              .ofType(String.class)
+              .defaultsTo("default-public-container");
+      ArgumentAcceptingOptionSpec<String> serviceId = parser.accepts("serviceId", "serviceId representing the caller")
+          .withOptionalArg()
+          .describedAs("serviceId")
+          .ofType(String.class)
+          .defaultsTo("PerfNettyClient");
```
Comment (vgkholla) [inline ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java]: minor: `NettyPerfClient`

File: ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java [hunk 4]
```diff
@@ -109,6 +112,9 @@
     final Long postBlobTotalSize;
     final Integer postBlobChunkSize;
     final String sslPropsFilePath;
+    final String targerAccountName;
```
Comment (vgkholla) [inline ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java]: typ: `targetAccountName`

File: ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java [hunk 5]
```diff
@@ -186,10 +212,15 @@ protected ClientArgs(String args[]) {
...
-      } else if (requestType.equals(POST) && (postBlobTotalSize == null || postBlobTotalSize <= 0
-          || postBlobChunkSize == null || postBlobChunkSize <= 0)) {
-        throw new IllegalArgumentException(
-            "Total size to be posted and size of each chunk need to be specified with POST and have to be > 0");
+      } else if (requestType.equals(POST)) {
+        if (postBlobTotalSize == null || postBlobTotalSize <= 0 || postBlobChunkSize == null
+            || postBlobChunkSize <= 0) {
+          throw new IllegalArgumentException(
+              "Total size to be posted and size of each chunk need to be specified with POST and have to be > 0");
+        } else if (targerAccountName == null || targetContainerName == null) {
```
Comment (vgkholla) [inline ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java]: service id check?

Comment (nsivabalan) [reply ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java]: we have defaults set for serviceId. hence null check may not make sense.

File: ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java [hunk 6]
```diff
@@ -186,10 +210,18 @@ protected ClientArgs(String args[]) {
...
+        if (postBlobTotalSize == null || postBlobTotalSize <= 0 || postBlobChunkSize == null
+            || postBlobChunkSize <= 0) {
+          throw new IllegalArgumentException(
+              "Total size to be posted and size of each chunk need to be specified with POST and have to be > 0");
+        } else if (targetAccountName == null || targetAccountName.isEmpty() || targetContainerName == null
+            || targetContainerName.isEmpty()) {
+          throw new IllegalArgumentException("TargetAccountName and TargetContainerName cannot be empty with POST");
+        }
+      }
+      if (serviceId == null || serviceId.isEmpty()) {
```
Comment (vgkholla) [inline ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java]: minor: `else if`

Comment (vgkholla) [reply ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java]: Ah actually never mind - missed the indentation

**Pull request merged**